libyul/Dialect.h

```diff
@@ -56,30 +59,33 @@ struct BuiltinFunction
 
 struct Dialect
 {
+	static size_t constexpr verbatimMaxInputSlots = 100;
+	static size_t constexpr verbatimMaxOutputSlots = 100;
+
 	/// Noncopiable.
 	Dialect(Dialect const&) = delete;
 	Dialect& operator=(Dialect const&) = delete;
 
-	/// @returns the builtin function of the given name or a nullptr if it is not a builtin function.
-	virtual BuiltinFunction const* builtin(YulName /*_name*/) const { return nullptr; }
+	/// @returns the builtin function of the given name or a null if it is not a builtin function.
+	virtual std::optional<BuiltinHandle> findBuiltin(std::string_view /*_name*/) const { return std::nullopt; }
+
+	virtual BuiltinFunction const& builtin(BuiltinHandle const&) const { yulAssert(false); }
```

Comment:
In addition to better names, it would not hurt to have these functions properly documented:
```suggestion
    /// Finds a builtin by name and returns the corresponding handle.
	/// @returns Builtin handle or null if the name does not match any builtin in the dialect.
	virtual std::optional<BuiltinHandle> findBuiltin(std::string_view /*_name*/) const { return std::nullopt; }

    /// Retrieves the description of a builtin function by its handle.
    /// Note that handles are dialect-specific and can be used only with a dialect that created them.
	virtual BuiltinFunction const& builtin(BuiltinHandle const&) const { yulAssert(false); }
```

Reply:
BTW, I see that `EVMDialect` repeats docstrings for some functions it overrides. We should remove them. They will easily go out of sync so it's unnecessary duplication. We should only have them if the overridden function actually needs some extra explanation.

Reply:
Could you also remove the docstring from `EVMDialect::reservedIdentifier()` too? That one's duplicated as well.